### ISSUE TITLE
"do some work" - server at port 7000 doesn't actually output JSON.

### ISSUE THREAD

**Issue by CamiloMM** [stevekane/promise-it-wont-hurt#25]
In this exercise, we're supposed to parse JSON from a server that is created at port `7000`. But it doesn't output JSON, instead it outputs GUIDs like this: `097a9790-1cf2-11e4-88f2-630a107fa5b3`. I have checked and I'm not running anything at that port (I don't get anything from that port if I'm not in the exercise).

> 1. Send an HTTP GET request to the session cache on port 7000.  **A JSON payload**
   will be returned to you containing a primary key called "id".

The only JSON parsing needed is for the second request:

```js
var qhttp = require('q-io/http');
qhttp.read('http://localhost:7000')
// .then(function (buffer) { return 'http://localhost:7001/' + JSON.parse(buffer.toString()).id; }) // SHOULD be the solution.
.then(function (buffer) { return 'http://localhost:7001/' + buffer.toString(); }) // NOT right. No JSON here.
.then(qhttp.read)
.post('toString')
.then(JSON.parse)
.then(console.log)
.done();
```

![screenshot of code running](https://cloud.githubusercontent.com/assets/2404138/3820683/6225702c-1cfc-11e4-9bb3-e00e3e93d5b3.png)

What's funny is the official solution *also* isn't consistent with the description of the task.

**Comment by dKab** [comment]
ohhh I've been trying to figure out  for hours what's wrong with this task! I couldn't guess that 097a9790-1cf2-11e4-88f2-630a107fa5b3 was the ID I needed, because I was looking for json...

**Comment by gbezyuk** [comment]
The issue is still open. I upvote it.

**Comment by TimothyGu** [comment]
This has been fixed by dbfc3cb4fd75cb71ae15511479fc5bedf3b28b22. We just need to publish the change.
